core/src/main/java/discord4j/core/object/entity/channel/ForumChannel.java

```diff
@@ -0,0 +1,247 @@
+
+    public ForumChannel(GatewayDiscordClient gateway, ChannelData data) {
+        super(gateway, data);
+    }
+
+    /**
+     * Gets the topic for this guild forum.
+     *
+     * @return the user defined topic for this guild forum channel (0-4096 characters).
+     */
+    public Optional<String> getTopic() {
+        return getData().topic().toOptional()
+            .flatMap(opt -> opt); // Value is present on guild forum type but nullable
+    }
+
+    /**
+     * Gets whether this channel is considered NSFW (Not Safe For Work).
+     *
+     * @return {@code true} if this channel is considered NSFW (Not Safe For Work), {@code false} otherwise.
+     */
+    public boolean isNsfw() {
+        return getData().nsfw().toOptional().orElse(false);
+    }
+
+    /**
+     * Gets the channels {@link Flag} associated to this forum channel
+     * Unknown flags are currently ignored.
+     *
+     * @return An {@link EnumSet} representing the <b>known flags</b> for this forum channel.
+     */
+    public EnumSet<Flag> getFlags() {
+        return getData().flags().toOptional()
+            .map(Flag::valueOf)
+            .orElseThrow(IllegalStateException::new); // Mandatory for Forum Channels
+    }
+
+    /**
+     * Gets the default auto archive duration for threads in this forum channel
+     *
+     * @return Default client auto archive duration in minutes
+     */
+    public int getDefaultAutoArchiveDuration() {
+        return getData().defaultAutoArchiveDuration().toOptional()
+            .orElseThrow(IllegalStateException::new); // Mandatory for Forum channels
+    }
+
+    /**
+     * Gets the available forum tags for this forum channel
+     *
+     * @return A list containing every available forum tag for this channel
+     */
+    public List<ForumTag> getAvailableTags() {
+        return getData().availableTags().toOptional()
+            .orElseThrow(IllegalStateException::new) // Mandatory for Forum channels
+            .stream()
+            .map(forumTagData -> new ForumTag(getClient(), forumTagData))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the default emoji to add in a reaction button on a thread for this forum channel
+     *
+     * @return An {@link Optional} that may contain the {@link DefaultReaction} emoji
+     */
+    public Optional<DefaultReaction> getDefaultReaction() {
+        return getData().defaultReactionEmoji()
+            .toOptional().orElseThrow(IllegalStateException::new) // Mandatory for Forum channels
+            .map(defaultReactionData -> new DefaultReaction(getClient(), defaultReactionData));
+    }
+
+    /**
+     * Gets the default thread rate limit duration per user.
+     * This field is copied into a created thread and is not updated on the fly.
+     *
+     * @return The initial rate limit per user for newly created threads
+     */
+    public int getDefaultThreadRateLimitPerUser() {
```

Comment:
Better make this optional because you dont have a form to check when this is set unless you get the data or use try/catch